### PR TITLE
Dashboard V2

### PR DESCRIPTION
**Pull request by jasonjoh**
This is a complete rewrite of the dashboard.

## Reasoning

1. Constant deluge of Dependabot PR's due to age of React front-end. Due to time/resource scarcity, testing and validating these PR's was not an easy task.
1. React front-end was using an old, deprecated auth library, oidc-connect. No longer maintained on GitHub, project is archived.
1. Backend was using outdated versions of the ASP.NET Core auth middleware.

## Improvements

1. Updated the auth stack to use msal-react and Microsoft.Identity.Web.
1. Recreated FE app, updating all NPM dependencies possible.
1. Simplified and cleaned up models. There was a lot of duplicated information, mix of conventions, multiple classes in a single file, etc.
1. Increased code re-use in React app by creating more reusable components.
1. Visual improvements (tweak to dark mode, user photo via Graph)

### CODE REVIEW

File: SamplesDashboard/SamplesDashboard/ClientApp/public/manifest.json [hunk 1]
```diff
@@ -1,14 +1,24 @@
 {
-  "short_name": "SamplesDashboard",
-  "name": "SamplesDashboard",
+  "short_name": "React App",
+  "name": "Create React App Sample",
```
Comment (MaggieKimani1) [inline SamplesDashboard/SamplesDashboard/ClientApp/public/manifest.json:3]: I think we should retain the name of the dashboard here 😊

File: SamplesDashboard/SamplesDashboard/Models/Repository.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        public int PullRequests { get; set; }
+        public int StarGazers { get; set; }
+        public int Views { get; set; }
+        public int Forks { get; set; }
+        public string Url { get; set; }
+        public string DefaultBranch { get; set; }
+        public DateTimeOffset LastUpdated { get; set; }
+        public string Language { get; set; }
+        public string FeatureArea { get; set; }
+        //public bool HasDependencies { get; set; }
```
Comment (MaggieKimani1) [inline SamplesDashboard/SamplesDashboard/Models/Repository.cs:25]: We can clean this commented code.

File: SamplesDashboard/SamplesDashboard/Services/GitHubAuthService.cs [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+                    new ProductHeaderValue(_configuration.GetValue<string>(Constants.Product),
+                                           _configuration.GetValue<string>(Constants.ProductVersion)))
+                {
+                    Credentials = new Credentials(jwtToken, AuthenticationType.Bearer)
+                };
+
+                // Get the installation ID of the app in the org
+                var installations = await gitHubClient.GitHubApps.GetAllInstallationsForCurrent();
+                var installId = installations
+                    .Where(i => i.Account.Login == _configuration.GetValue<string>(Constants.GitHubOrg))
```
Comment (MaggieKimani1) [inline SamplesDashboard/SamplesDashboard/Services/GitHubAuthService.cs:79]: This can be simplified to

**Review by MaggieKimani1 — CHANGES_REQUESTED** [review]

**Comment by jasonjoh** [comment]
So somehow when pulling updates from here into my fork, it merged? I'm not sure how that happened.

**Pull request merged**